File: ZYSwiftUIFrame/ZYSwiftUIFrame/UI/Meeting/Meeting.swift
//
//  Meeting.swift
//  ZYSwiftUIFrame
//
//  Created by lzy on 2022/3/31.
//

/**
 * 会议实体类
 */
struct Meeting: Codable, Identifiable, Equatable {
    static func == (lhs: Meeting, rhs: Meeting) -> Bool {
        return lhs.mtName == rhs.mtName
        && lhs.mtSummary == rhs.mtSummary
        && lhs.mtTheme == rhs.mtTheme
        && lhs.mtTime == rhs.mtTime
    }
    
    var id: Int? // id 可以为 nil 是为了区分 新增 和 更新
    @Default<String> var mtName: String
    @Default<String> var mtTheme: String
    @Default<String> var mtSummary: String
    @Default<String> var mtContent: String
    @Default<String> var mtMember: String
    @Default<String> var mtTime: String
    @Default<String> var remark: String
    @Default<Int> var createUser: Int
    
    var anList: [Upload] // 附件列表
    // 新增参数
    var annexIds: [Int]? // 附件Id数组
    
    init() {
        self.mtName = ""
        self.mtTheme = ""
        self.mtSummary = ""
        self.mtContent = ""
        self.mtMember = ""
        self.mtTime = ""
        self.remark = ""
        self.createUser = -1
        self.anList = []
        
        self.annexIds = []
    }

}

File: go_api_server/model/common/request/meeting.go
package request

type ChangeMeetingInfo struct {
	ID         uint   `json:"id" gorm:"primarykey"`
	MtName     string `json:"mtName" gorm:"comment:会议名称"`
	MtTheme    string `json:"mtTheme" gorm:"comment:会议主题"`
	MtSummary  string `json:"mtSummary" gorm:"comment:会议概要"`
	MtContent  string `json:"mtContent" gorm:"comment:会议内容"`
	MtMember   string `json:"mtMember" gorm:"comment:参会人员"`
	MtTime     string `json:"mtTime" gorm:"comment:开会时间"`
	CreateUser int    `json:"createUser" gorm:"comment:创建人"`
	Remark     string `json:"remark" gorm:"comment:备注"` // 备注
	// 1 个会议可以有多个 图片
	AnnexIds []int `json:"annexIds"`
}
